### PR TITLE
docs(agw): [proposal] UE Initiated PDU Session Modification

### PR DESCRIPTION
**Pull request by GANESH-WAVELABS**
[proposal] UE Requested PDU Session Modification with magma

## Summary
Handling of UE Requested PDU Session Modification in magma
Corresponding Github issue : https://github.com/magma/magma/issues/7971

## Test Plan
UE requested session modification should be tested with DSTester and TeraVM.

## Additional Information
Signed-off-by: GANESH-WAVELABS <ganesh.irrinki@wavelabs.ai>

### CODE REVIEW

File: docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+   running, the UE shall initiate the UE-requested PDU session modification
+   procedure after expiry of timer T3396. 
+
+4. When the UE-requested PDU session modification procedure is used to indicate
+   to request specific QoS handling and segregation of service data flows 
+
+## **Use Case**
+
+The UE can request the network to modify or release PDU sessions. The network
+can fulfil such a request from the UE by modifying a PDU session or releasing
```
Comment (electronjoe) [inline docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md]: Minor misspelling of `fulfill`

**Review by electronjoe — CHANGES_REQUESTED** [review]

**Comment by electronjoe** [comment]
Is there an associated Github Issue for this proposal? The process calls for one, it's nice to link it in the PR description so they are tied.

File: docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md (hunk 1 above)
Comment (GANESH-WAVELABS) [reply docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md]: Corrected the misspelling. Thank you

**Comment by GANESH-WAVELABS** [comment]
> Is there an associated Github Issue for this proposal? The process calls for one, it's nice to link it in the PR description so they are tied.

Added Corresponding Github issue : #7971 in Summary.

**Comment by uri200** [comment]
Please, plan on adding unit test to those systems that allow it currently. For example sessiond.

Also what is the plan to test it with TereaVm?  What is DSTest?

File: docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+       --  UE Integrity Protection Maximum Data Rate 
+
+     The NAS message is forwarded by (R)AN to AMF with indication of user
+     location information. AMF calls Nsmf_PDUSession_UpdateSMContext (SM context
+     ID, N1 SM container (PDU session modification request)). 
+
+2. SMF responds AMF after updating SM_Context and SMF requests to transfer
+   downlink N2 message to the UE and AN through the AMF. i.e. The SMF request
+   to update activation of Data Radio Bearer and N3 tunnel for the PDU session. 
+
```
Comment (uri200) [inline docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md:121]: When SMF receives the request to modify Qos, how does SMF decides if this change is valid or not? 
Does SMF stores any kind of profile information ? Does SMF do a query to any other system to check if the Qos change is allowed?

**Comment by ulaskozat** [comment]
@sureshgorijavolu can you link the zenhub issue for UE initiated session modification to this proposal?

**Comment by hcgatewood** [comment]
@uri200 @ulaskozat where are we at with this PR

**Comment by ulaskozat** [comment]
@hcgatewood it is still under progress and discussion.

**Comment by GANESH-WAVELABS** [comment]
As per Phil's suggestion "this feature to be suspended as there is no demand from the customers for years on this features" Hence take it off the radar closing the issue.

**Pull request closed without merging**